generators/pkg/generator/generate_test.go

```diff
@@ -6,33 +6,285 @@ import (
+				{
+					Name:        "deleteExisting",
+					Type:        v1beta1.ParamTypeString,
+					Description: "clean out the contents of the repo's destination directory (if it already exists) before trying to clone the repo there",
+					Default: &v1beta1.ArrayOrString{
+						Type:      v1beta1.ParamTypeString,
+						StringVal: "false",
+					},
+				},
+				{
+					Name:        "httpProxy",
+					Type:        v1beta1.ParamTypeString,
+					Description: "git HTTP proxy server for non-SSL requests",
+					Default: &v1beta1.ArrayOrString{
+						Type:      v1beta1.ParamTypeString,
+						StringVal: "",
+					},
+				},
+				{
+					Name:        "httpsProxy",
+					Type:        v1beta1.ParamTypeString,
+					Description: "git HTTPS proxy server for SSL requests",
+					Default: &v1beta1.ArrayOrString{
+						Type:      v1beta1.ParamTypeString,
+						StringVal: "",
+					},
+				},
+				{
+					Name:        "noProxy",
+					Type:        v1beta1.ParamTypeString,
+					Description: "git no proxy - opt out of proxying HTTP/HTTPS requests",
+					Default: &v1beta1.ArrayOrString{
+						Type:      v1beta1.ParamTypeString,
+						StringVal: "",
+					},
+				},
+			},
+			Results: []v1beta1.TaskResult{
+				{
+					Name:        "commit",
+					Description: "The precise commit SHA that was fetched by this Task",
+				},
+			},
+			Steps: []v1beta1.Step{
+				{
+					Container: corev1.Container{
+						Name:  "clone",
+						Image: "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init:v0.13.2",
+					},
+					Script: "CHECKOUT_DIR=\"$(workspaces.output.path)/$(params.subdirectory)\"\n\ncleandir() {\n  # Delete any existing contents of the repo directory if it exists.\n  #\n  # We don't just \"rm -rf $CHECKOUT_DIR\" because $CHECKOUT_DIR might be \"/\"\n  # or the root of a mounted volume.\n  if [[ -d \"$CHECKOUT_DIR\" ]] ; then\n    # Delete non-hidden files and directories\n    rm -rf \"$CHECKOUT_DIR\"/*\n    # Delete files and directories starting with . but excluding ..\n    rm -rf \"$CHECKOUT_DIR\"/.[!.]*\n    # Delete files and directories starting with .. plus any other character\n    rm -rf \"$CHECKOUT_DIR\"/..?*\n  fi\n}\n\nif [[ \"$(params.deleteExisting)\" == \"true\" ]] ; then\n  cleandir\nfi\n\ntest -z \"$(params.httpProxy)\" || export HTTP_PROXY=$(params.httpProxy)\ntest -z \"$(params.httpsProxy)\" || export HTTPS_PROXY=$(params.httpsProxy)\ntest -z \"$(params.noProxy)\" || export NO_PROXY=$(params.noProxy)\n\n/ko-app/git-init \\\n  -url \"$(params.url)\" \\\n  -revision \"$(params.revision)\" \\\n  -refspec \"$(params.refspec)\" \\\n  -path \"$CHECKOUT_DIR\" \\\n  -sslVerify=\"$(params.sslVerify)\" \\\n  -submodules=\"$(params.submodules)\" \\\n  -depth \"$(params.depth)\"\ncd \"$CHECKOUT_DIR\"\nRESULT_SHA=\"$(git rev-parse HEAD | tr -d '\\n')\"\nEXIT_CODE=\"$?\"\nif [ \"$EXIT_CODE\" != 0 ]\nthen\n  exit $EXIT_CODE\nfi\n# Make sure we don't add a trailing newline to the result!\necho -n \"$RESULT_SHA\" > $(results.commit.path)\n",
```

Comment:
We can make the Script field easier to read using raw Strings using `. See https://github.com/tektoncd/pipeline/blob/master/pkg/pod/pod_test.go#L485